### PR TITLE
Enable cdash appveyor

### PR DESCRIPTION
**Pull request by robertmaynard**
Extending on the Travis support, this makes the AppVeyor builds report to CDash.

@salasoom Should I be expecting all the AppVeyor builds to work? I am currently seeing compiler failures when `OpenMP` is enabled.

### CODE REVIEW

**Comment by salasoom** [comment]
I fixed the typo with passing `CFP_NAMESPACE` into the build, and changed the way "cfp" and "namespace" get added to the test site name (for clarity).
Also related, in the travis/cdash PR #46 I updated how the 2 CI builds are configured. My commit on this branch has updated Appveyor to do the new 2 build config.

**Comment by salasoom** [comment]
I rebased this branch against #46 to pull in my CI changes because the Travis build was failing here without it. The changes I pulled in covered what was in my new commit - that's why it's gone now.

File: appveyor.yml [hunk 1]
```diff
@@ -74,25 +74,12 @@ build_script:
...
 
-  # build/test with OpenMP, with CFP custom namespace
-  - if "%COMPILER%"=="msvc" cmake -G "%GENERATOR%" -DCMAKE_BUILD_TYPE="%BUILD_TYPE%" -DZFP_ENDTOEND_TESTS_ONLY=1 -DBUILD_CFP=ON -DCFP_NAMESPACE=cfp2 ..
-  - if not "%COMPILER%"=="msvc" cmake -G "%GENERATOR%" -DCMAKE_BUILD_TYPE="%BUILD_TYPE%" -DCMAKE_SH=CMAKE_SH-NOTFOUND -DZFP_ENDTOEND_TESTS_ONLY=1 -DBUILD_CFP=ON -DCFP_NAMESPACE=cfp2 ..
-
-  - if "%COMPILER%"=="msvc" cmake --build . --config "%BUILD_TYPE%"
-  - if not "%COMPILER%"=="msvc" cmake --build .
-
-  # only run tests not run in previous build, due to appveyor time limit (1 hour)
-  - ctest -V -C "%BUILD_TYPE%" -R ".*Omp.*"
```
Comment (salasoom) [inline appveyor.yml]: For the 2nd build, the regex seems to have been dropped. Can we restore this behavior? All tests are run for the 1st build, but for the 2nd build, only run tests matching regex ".*Omp.*"

Comment (robertmaynard) [reply appveyor.yml]: I will look into how we can add this in.

Comment (robertmaynard) [reply appveyor.yml]: Okay updated to only run the OpenMP tests for the second build

**Comment by salasoom** [comment]
@robertmaynard I've made some changes to the Appveyor build behavior because there were holes in the behavior that I was unaware of. Now that's all finished, but I have one more question:

I periodically see Appveyor builds fail because of an HTTP 443 timeout when trying to send the results to CDash. I can restart the build and have it pass the 2nd time. Is this something you've seen before? Unfortunately I don't have the log saved because I restarted it and they all pass now.

**Comment by salasoom** [comment]
@robertmaynard The latest build showed 2 different errors when trying to send data to CDash (shown in Appveyor's logs). Some quick googling suggests this is a problem on Appveyor's side. I asked you about it in case you had run into this before and happen to know of a quick fix. If not, don't sweat it.

Submit files (using https)
   Send to track: AppVeyor
   Using HTTP submit method
   Drop site:https://open.cdash.org/submit.php?project=zfp
   Error when uploading file: C:/projects/zfp/build/Testing/20190313-1410/Test.xml
   Error message was: schannel: next InitializeSecurityContext failed: SEC_E_MESSAGE_ALTERED (0x8009030F) - The message or signature supplied for verification has been altered
   Problems when submitting via HTTP

Submit files (using https)
   Send to track: AppVeyor
   Using HTTP submit method
   Drop site:https://open.cdash.org/submit.php?project=zfp
   Error when uploading file: C:/projects/zfp/build/Testing/20190313-0908/Test.xml
   Error message was: Failed to connect to open.cdash.org port 443: Timed out
   Problems when submitting via HTTP

Otherwise, I'm assuming all the CDash/Appveyor functionality is in place, and I will go ahead and merge this PR.

**Comment by robertmaynard** [comment]
I haven't seen that issue before but I will ask other CDash experts

**Pull request merged**